overlord/snapstate/snapstate_test.go

```diff
@@ -5419,6 +5420,51 @@ func (s *snapmgrTestSuite) TestEnsureRefreshRefusesWeekdaySchedules(c *C) {
 	c.Check(logbuf.String(), testutil.Contains, `cannot use refresh.schedule configuration: cannot parse "mon@12:00": not a valid time`)
 }
 
+func (s *snapmgrTestSuite) TestEnsureRefreshLegacyScheduleIsLowerPriority(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+	snapstate.CanAutoRefresh = func(*state.State) (bool, error) { return true, nil }
+
+	logbuf, restore := logger.MockLogger()
+	defer restore()
+
+	s.state.Set("last-refresh", time.Date(2009, 8, 13, 8, 0, 5, 0, time.UTC))
+	tr := config.NewTransaction(s.state)
+	tr.Set("core", "refresh.timer", fmt.Sprintf("00:00-23:59,,mon,12:00-14:00"))
+	// legacy schedule is invalid
+	tr.Set("core", "refresh.schedule", fmt.Sprintf("00:00-23:59/mon@12:00-14:00"))
+	tr.Commit()
+
+	// Ensure() also runs ensureRefreshes()
+	s.state.Unlock()
+	s.snapmgr.Ensure()
+	s.state.Lock()
+
+	// expecting no log that the schedule cannot be parsed
+	c.Check(logbuf.String(), Not(testutil.Contains), `cannot use refresh.schedule configuration: cannot parse "mon@12:00": not a valid time`)
```

Comment:
Can we have a stronger check here? Log messages tend to get tweaks over time and I feel that if this message gets tweaked in the code (without updating this test check) and later we break schedule priorities, this check will never catch the breakage. Would it be possible to check what schedule was actually used? In other words a positive check would be better here.

Reply:
Very good point. A test for the presence of a log message is fine, but a test for the lack of an error message is weak.